src/stk/molecular/topology_graphs/host_guest/complex.py

```diff
@@ -133,36 +186,98 @@ def __init__(
             If `guest_start` or `guest_target` is defined but the other
             is not.
 
+        :class:`ValueError`
+            If the old API and new API are used simultaneously.
+
         """
 
-        num_nones = sum(
-            1 for vector in (guest_start, guest_target)
-            if vector is None
-        )
-        if num_nones == 1:
-            raise TypeError(
-                'If guest_start or guest_target is defined, '
-                'the other must be too.'
+        if guests is not None and guest is not None:
+            raise ValueError(
+                'You are attempting to use the old API with the new '
+                'API. Either use the stk.host_guest.Guest API and '
+                '`guests`, or the old API.'
```

Comment:
Do not suggest that they used the old API. Suggest that they use the `guests` parameter explicitly